### ISSUE TITLE
[Bug] Unable to destroy carts

### ISSUE THREAD

**Issue by MikeOnDeck** [constellation-mc/Linkart#25]
**Describe the bug**
Occasionally, a Minecart may end up just swaying more agreesively after each hit but it never breaks

**To Reproduce**
1. Place down several minecarts
2. Link them
3. Try to destroy (seemingly more repo'able when destroying from back
4. Unable to destroy a cart, see the cart sway more than normal

**Expected behavior**
The minecart should destroy as normal after several hits (punches)

**Environment:**
 - Minecraft Version [1.20.1]
 - Mod Version [5.4.1 (1.19.2-1.20.4) fabric

Uploading IMG_1055.mov…

**Comment by zenfyrdev** [comment]
Hello, please send me your game logs. Usually every hit is followed by a crash dump.

**Comment by MikeOnDeck** [comment]
[latest.log](https://github.com/melontini/Linkart/files/14415028/latest.log)

I think this is the one. It's from my headless server (running on a Pi 5)

**Comment by zenfyrdev** [comment]
I don't think I can reproduce the issue, but can you try this build? https://github.com/melontini/Linkart/actions/runs/8084039527

**Comment by MikeOnDeck** [comment]
Hey @melontini, it looks like your newer build might have fixed the bug actually. I tried the old build, managed to repro it again pretty quickly, then tried the new and haven't managed to get it to happen again since. Thanks so much for providing this!

I did actually manage to repro a different (less severe) issue with that build though. I was thinking of creating a new bug for it, but since that's a build you provided maybe I shouldn't? I'm happy to make one if that works better for you, but the bug itself is as follows:
1. Place down 3 carts - Cart A, B and C
2. Press link on Cart A, then link Cart B.
3. Cart A and B are linked. (Cart A is the leader, B is the follower).
4. Press link on Cart C, then link Cart B by clicking on it with the chain.
5. Cart B is now paired to Cart C, but you have consumed a new chain.
6. Even after removing all the carts, the first chain you used is forever consumed.
